File: frogment/src/main/java/ch/grze/frogment/core/component/AbstractActivityComponentInjector.java
package ch.grze.frogment.core.component;

import android.app.Activity;

import ch.grze.frogment.core.Core;

public abstract class AbstractActivityComponentInjector {
    abstract public void inject(Core core, Activity activity);

    protected <T> T getTypedActivity(Activity activity) throws ClassCastException {
        return (T) activity;
    }
}

File: frogment/src/main/java/ch/grze/frogment/core/component/AbstractActivityComponentInjector.kt
package ch.grze.frogment.core.component

import android.app.Activity

import ch.grze.frogment.core.Core

@Deprecated("fallback, use ComponentInjector")
abstract class AbstractActivityComponentInjector : ComponentInjector<Activity> {
    @Throws(ClassCastException::class)
    protected fun <T> getTypedActivity(activity: Activity): T = activity as T
}
